openid-connect-server/src/main/java/org/mitre/openid/connect/web/ClientAPI.java

```diff
@@ -244,6 +245,11 @@ public String apiAddClient(@RequestBody String jsonString, Model m, Authenticati
 			m.addAttribute(HttpCodeView.CODE, HttpStatus.BAD_REQUEST);
 			m.addAttribute(JsonErrorView.ERROR_MESSAGE, "Unable to save client: " + e.getMessage());
 			return JsonErrorView.VIEWNAME;
+		} catch (PersistenceException e) {
```

Comment:
Nice catch. I added checks to make sure that the cause was from a duplicate client id entry.